File: src/java/org/javaturk/dp/ch03/ocp/factory/solution/EmployeeFactory.java
package org.javaturk.dp.ch03.ocp.factory.solution;

import org.javaturk.dp.ch04.factoryMethod.employee.EmployeeRandomizer;

public class EmployeeFactory implements Factory {

	@Override
	public Employee create() {
		Employee employee = new Employee(EmployeeRandomizer.createId(), EmployeeRandomizer.createName(),
				EmployeeRandomizer.createYear(), EmployeeRandomizer.createDepartment());
		return employee;
	}

}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/factory/employee1/employee-factory.ts
import {Factory} from "./factory";
import {Employee} from "./employee";
import {EmployeeRandomizer} from "../employee-randomizer";

export class EmployeeFactory implements Factory {
  create(): Employee {
    return new Employee(EmployeeRandomizer.createId(), EmployeeRandomizer.createName(),
        EmployeeRandomizer.createYear(), EmployeeRandomizer.createDepartment());
  }
}